ext/rotoscope/rotoscope.c

```diff
@@ -126,20 +89,41 @@ VALUE escape_csv_string(VALUE string) {
   return rb_funcall(string, id_gsub, 2, str_quote, str_escaped_quote);
 }
 
-static void log_trace_event_with_caller(VALUE output_buffer, VALUE io,
-                                        rs_stack_frame_t *stack_frame,
-                                        rs_stack_frame_t *caller_frame) {
-  VALUE escaped_method_name = escape_csv_string(stack_frame->tp.method_name);
-  VALUE escaped_caller_method_name =
-      escape_csv_string(caller_frame->tp.method_name);
+typedef struct {
+  VALUE class_name;
+  VALUE name;
+  const char *level;
+} rs_method_log_fields_t;
+
+static rs_method_log_fields_t method_log_fields(rs_method_desc_t *method) {
+  if (method == NULL) {
+    return (rs_method_log_fields_t){
+        .class_name = str_unknown_class_name,
```

Comment:
Why does `str_unknown_class_name` and `str_unknown_method_name` needs to return a Qvalue, but `level` returns a C string ?

Reply:
The class and method names come from ruby, so they are ruby objects (i.e. have type VALUE).  We want to hold onto the VALUE on the stack rather than just the character pointer so that ruby doesn't garbage collect them.

The `level` always references string literals, so we can just use a pointer to them without having to worry about freeing them.